packages/suite/src/reducers/wallet/fiatRateReducer.ts

```diff
@@ -1,35 +1,42 @@
-import { RATE_UPDATE, FiatRateActions } from '@wallet-middlewares/coingeckoMiddleware';
 import produce from 'immer';
+import { Action } from '@suite-types';
+import { RATE_UPDATE } from '@wallet-actions/constants/fiatRatesConstants';
+import { STORAGE } from '@suite-actions/constants';
+import { Network } from '@wallet-types';
 
 export interface Fiat {
-    symbol: string;
+    symbol: Network['symbol'];
     rates: { [key: string]: number };
+    timestamp: number;
 }
 
 export const initialState: Fiat[] = [];
 
-const update = (state: Fiat[], action: FiatRateActions) => {
-    const { symbol, rates } = action;
+const update = (state: Fiat[], fiat: Pick<Fiat, 'symbol' | 'rates'>) => {
+    const { symbol, rates } = fiat;
     const affected = state.find(f => f.symbol === symbol);
-
     Object.keys(rates).map(k => rates[k].toFixed(2));
     if (!affected) {
         state.push({
             symbol,
             rates,
+            timestamp: Date.now(),
```

Comment:
https://redux.js.org/style-guide/style-guide#reducers-must-not-have-side-effects